File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Go/s221078931_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
	"sort"
)

const MaxInt = 2147483647

type IntegralRect struct {
	Height int
	Width  int
}

func (r *IntegralRect) diagonal() int {
	if r.Height >= r.Width {
		return MaxInt
	}
	return r.Height*r.Height + r.Width*r.Width
}

func main() {
	rects := make([]IntegralRect, 0, 40000)
	for i := 0; i < 200; i++ {
		for j := 0; j < 200; j++ {
			h := i + 1
			w := j + 1
			rects = append(rects, IntegralRect{Height: h, Width: w})
		}
	}

	sort.Slice(rects, func(i, j int) bool {
		a := &rects[i]
		b := &rects[j]
		d1 := a.diagonal()
		d2 := b.diagonal()
		if d1 != d2 {
			return d1 < d2
		}
		return a.Height < b.Height
	})

	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		line := scanner.Text()
		fields := strings.Fields(line)
		if len(fields) != 2 {
			continue
		}
		hStr := fields[0]
		wStr := fields[1]
		h, _ := strconv.Atoi(hStr)
		w, _ := strconv.Atoi(wStr)
		if h == 0 && w == 0 {
			break
		}
		tmp := IntegralRect{Height: h, Width: w}
		diag := tmp.diagonal()
		found := false
		for i := 0; i < len(rects); i++ {
			r := rects[i]
			d1 := r.diagonal()
			h1 := r.Height
			h2 := tmp.Height
			if (d1 > diag) || (d1 == diag && h1 > h2) {
				fmt.Printf("%d %d\n", r.Height, r.Width)
				found = true
				break
			}
		}
	}
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/C++/s221078931_commented.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <algorithm>
#include <climits> // Added to use INT_MAX
using namespace std;

struct IntegralRect {
    int height, width;
    IntegralRect(int h, int w) : height(h), width(w) {}
    int diagonal() const {
        if (height >= width) {
            return INT_MAX;
        }
        return height * height + width * width;
    }
    bool operator<(const IntegralRect& other) const {
        int d1 = diagonal();
        int d2 = other.diagonal();
        if (d1 != d2) return d1 < d2;
        return height < other.height;
    }
};

int main() {
    IntegralRect rects[40000];
    for (int i = 0; i < 200; ++i) {
        for (int j = 0; j < 200; ++j) {
            rects[i * 200 + j] = IntegralRect(i + 1, j + 1);
        }
    }
    sort(rects, rects + 40000);

    while (true) {
        string line;
        if (!getline(cin, line)) break;
        istringstream iss(line);
        int h, w;
        iss >> h >> w;
        if (h == 0 && w == 0) break;
        IntegralRect tmp(h, w);
        for (int i = 0; i < 40000; ++i) {
            if (tmp < rects[i]) {
                cout << rects[i].height << " " << rects[i].width << endl;
                break;
            }
        }
    }
    return 0;
}
// End of Code
